programs/vote/src/vote_state/mod.rs

```diff
@@ -667,16 +658,25 @@ pub fn process_vote_unchecked(vote_state: &mut VoteStateV3, vote: Vote) -> Resul
 }
 
 #[cfg(test)]
-pub fn process_slot_votes_unchecked(vote_state: &mut VoteStateV3, slots: &[Slot]) {
+pub fn process_slot_votes_unchecked(vote_state: &mut VoteStateHandler, slots: &[Slot]) {
     for slot in slots {
         process_slot_vote_unchecked(vote_state, *slot);
     }
 }
 
-pub fn process_slot_vote_unchecked(vote_state: &mut VoteStateV3, slot: Slot) {
+pub fn process_slot_vote_unchecked(vote_state: &mut VoteStateHandler, slot: Slot) {
     let _ = process_vote_unchecked(vote_state, Vote::new(vec![slot], Hash::default()));
 }
 
+// Temporary adapter test method for runtime, core, etc. while vote state
+// transition to handler is underway.
+#[cfg(feature = "dev-context-only-utils")]
+pub fn process_slot_vote_unchecked_v3_for_tests(vote_state_v3: &mut VoteStateV3, slot: Slot) {
```

Comment:
The need for this is probably a decent reason to introduce a `VoteStateHandle` trait actually. If we had the `VoteStateHandle` trait implemented for both `VoteStateHandler` and `VoteStateV3`, then passing either of those structs to `process_slot_vote_unchecked` would be fine. This same issue will apply for v4 too